bindgen-tests/tests/tests.rs

```diff
@@ -427,6 +427,49 @@ fn test_header_contents() {
     assert_eq!(expected, actual);
 }
 
+const HIDDEN_FUNCTIONS_HEADER: &str = r#"
+__attribute__((visibility("hidden"))) int hidden_fn(void);
+__attribute__((visibility("protected"))) int protected_fn(void);
+int visible_fn(void);
+"#;
+
+fn hidden_functions_builder() -> Builder {
+    builder()
+        .allowlist_function("visible_fn")
+        .allowlist_function("hidden_fn")
+        .allowlist_function("protected_fn")
+        .header_contents("hidden.h", HIDDEN_FUNCTIONS_HEADER)
+}
```

Comment:
Please reuse the existing test framework instead. I.e. add a header to `bindgen-tests/tests/headers` and the generated file to `bindgen-tests/tests/expectations`. That way we can test the whole content.